internal/task/manager/manager.go

```diff
@@ -0,0 +1,68 @@
+package task
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	basetask "github.com/mikewurtz/taskman/internal/task"
+)
+
+type TaskManager struct {
+	mu sync.RWMutex
+	// task map by task ID
+	tasksMapByID map[string]*Task
+	ctx          context.Context
+}
+
+type Task struct {
+	mu                sync.RWMutex
+	ID                string
+	ClientID          string
+	ProcessID         int
+	Status            int
+	StartTime         time.Time
+	ExitCode          *int32
+	TerminationSignal string
+	TerminationSource string
+	EndTime           time.Time
+	done              chan struct{}
+	doOnce            sync.Once
+}
```

Comment:
Given that the fields that the mutex guards are public, are we confident we don't have a data race here?

Reply:
Will get this updated to private with accessors